NEMbox/player.py

```diff
@@ -180,7 +180,10 @@ def append_songs(self, datalist):
             else:
                 database_song = self.songs[str(song["song_id"])]
                 if (database_song["song_name"] != song["song_name"]
-                    or database_song["quality"] != song["quality"]):
+                    or database_song["quality"] != song["quality"]
+                    or database_song["mp3_url"] != song["mp3_url"]):
+                    if "cache" in self.songs[str(song["song_id"])].keys():
+                        song["cache"] = self.songs[str(song["song_id"])]["cache"]
```

Comment:
@Catofes 
this style may be better.
btw, I really really really really like your avatar!!!!